src/main/java/com/google/api/generator/gapic/composer/MethodSampleCodeHelperComposer.java

```diff
@@ -118,10 +117,46 @@ private static AssignmentExpr assignClientVariableWithCreateMethodExpr(
       Map<String, ResourceName> resourceNames) {
     return arguments.stream()
         .map(
-            arg ->
-                createAssignmentExpr(
-                    argVarExprs.get(arg.name()),
-                    DefaultValueComposer.createDefaultValue(arg, resourceNames)))
+            arg -> {
+              Expr defaultValueExpr =
+                  DefaultValueComposer.createDefaultValue(arg, resourceNames, true);
```

Comment:
Here: If `arg` is String-typed and it has a matching resource name, pass that arg in instead.